benchmark/httpapi.py

```diff
@@ -277,18 +433,23 @@ def main(reactor, args):
         sys.stderr.write(options.getUsage())
         raise SystemExit(1)
 
+    def get_backend_from_options(options):
+        backend = _BACKENDS[options['backend']]
+
+        conn = dict()
+        if options['db-hostname']:
+            conn['hostname'] = options['db-hostname']
+        if options['db-port']:
+            conn['port'] = options['db-port']
+
+        return backend(**conn)
+
     startLogging(sys.stderr)
-    service = create_api_service(TCP4ServerEndpoint(reactor, options['port']))
 
-    # XXX Make startService() raise an exception on an error
-    # instead of just logging and dropping it.
```

Comment:
Actually the comment is about the next line of the code, about something we do already but in a somewhat ugly way:
```.py
    service._raiseSynchronously = True
```
Probably the comment could use some improvement to make that clearer.